cmake/modules/yaml.cmake

```diff
@@ -364,22 +426,50 @@ function(yaml_save)
   if(NOT yaml_file)
     zephyr_check_arguments_required(${CMAKE_CURRENT_FUNCTION} ARG_YAML FILE)
   endif()
-
-  zephyr_get_scoped(json_content ${ARG_YAML_NAME} JSON)
-  to_yaml("${json_content}" 0 yaml_out)
-
   if(DEFINED ARG_YAML_FILE)
     set(yaml_file ${ARG_YAML_FILE})
   else()
     zephyr_get_scoped(yaml_file ${ARG_YAML_NAME} FILE)
   endif()
+
+  zephyr_get_scoped(genex ${ARG_YAML_NAME} GENEX)
+  if(genex AND NOT ARG_YAML_TARGET)
+    message(WARNING "YAML context '${ARG_YAML_NAME}' uses generator expressions "
+                    "which require a TARGET parameter to 'yaml_save()'. GENEX "
+                    "entries will be included as comments in the output."
+    )
+  endif()
+
+  zephyr_get_scoped(json_content ${ARG_YAML_NAME} JSON)
+  to_yaml("${json_content}" 0 yaml_out ${genex})
+
   if(EXISTS ${yaml_file})
     FILE(RENAME ${yaml_file} ${yaml_file}.bak)
   endif()
   FILE(WRITE ${yaml_file} "${yaml_out}")
+
+  if (genex AND ARG_YAML_TARGET)
+    cmake_path(SET yaml_path "${yaml_file}")
+    cmake_path(GET yaml_path STEM yaml_file_no_ext)
+    set(json_file ${yaml_file_no_ext}.tmp)
+
+    FILE(GENERATE OUTPUT ${json_file}
+      CONTENT "${json_content}"
+      TARGET ${ARG_YAML_TARGET}
```

Comment:
I think we can remove this as you are creating a custom target below which seems to be sufficient to support the need for genex'es.

Unless that there are use-cases unknown to me, so feel free to reply :slightly_smiling_face: .